textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -23,7 +23,7 @@
  * @author laishuying
  */
 public class RegexMatcher implements IOperator {
-    private final IPredicate predicate;
+    private final RegexPredicate regexPredicate;
```

Comment:
I am ok with exposing to outside clients as `IPredicate` and maintaining internally as `RegexPredicate`. This is what we are doing here. The `RegexMatcher` constructor accepts `IPredicate`, but the class internally maintains it as `RegexPredicate`